cfme/utils/appliance/__init__.py

```diff
@@ -1012,17 +1012,25 @@ def ssh_client(self):
             }
         if self.is_dev:
             connect_kwargs.update({'is_dev': True})
-        ssh_client = ssh.SSHClient(**connect_kwargs)
-        try:
-            ssh_client.get_transport().is_active()
-            logger.info('default appliance ssh credentials are valid')
-        except Exception as e:
-            if self.is_dev:
-                raise Exception('SSH access on a dev alliance, (unsupported)')
-            logger.error(e)
-            logger.error('default appliance ssh credentials failed, trying establish ssh connection'
-                         ' using ssh private key')
-            ssh_client = self.ssh_client_with_privatekey()
+
+        def create_ssh_connection():
+            ssh_client = ssh.SSHClient(**connect_kwargs)
+            try:
+                ssh_client.get_transport().is_active()
+                logger.info('default appliance ssh credentials are valid')
+            except Exception as e:
+                if self.is_dev:
+                    raise Exception('SSH access on a dev appliance (unsupported): {}'.format(e))
+                logger.exception(
+                    ('default appliance ssh credentials failed: {}, '
+                     'trying establish ssh connection using ssh private key').format(e))
+                ssh_client.close()
+
+                ssh_client = self.ssh_client_with_privatekey()
+            assert ssh_client.run_command('true').success
+            return ssh_client
+
+        ssh_client = wait_for(func=create_ssh_connection, delay=5, timeout=120).out
```

Comment:
I don't see handing assert

Reply:
There's some handling of the exceptions within `create_ssh_connection`, so I think this wait-retry is still functional, just limited in scope for what will be handled and what will immediately fail.